### PR TITLE
Refactoring 5

### LINKED ISSUE #442: 🚀 [Feature Request] - Transition to new HomePage
### Is your feature request related to a problem ? Please describe. ✍️

We will be moving to the new HomePage...
new theme is: https://grabtern-new-frontend.vercel.app/ ( light theme)

- We have card now ( login card ) we will remove that.. infact clicking on signIn button open a page ( which is currently developed )

source code you can take help : https://github.com/anmode/grabtern-NewFrontend ( tailwind classes are there you can copy directly)
Our theme color new is `#845ec2`
Remember background should not be pure white* See the new theme

### Describe briefly how you would implement this enhancement.

```bash
Null
```

### Additional context or screenshots

Additional context

### Are you contributing for a program?

GSSoC2023

### Code of Conduct

- [X] I have read and agree to the Contributing Guidelines of this project.

### PR DESCRIPTION
**Pull request by Isha988**
## Related Issue
Follow up for  #498 
Closes #442 

## Description of Changes
1. Fix issue left in refactoring pr 4
2. Transition to new home page.

### CODE REVIEW

**Comment by anmode** [comment]
ek choti fix rh gyi... please could you look out:
https://user-images.githubusercontent.com/98320971/253114902-c11482a6-6c9e-4112-80d0-0a6c76df1643.png
Ig woh 4 mentors wala column rh gya h

**Comment by Isha988** [comment]
> ek choti fix rh gyi... please could you look out: https://user-images.githubusercontent.com/98320971/253114902-c11482a6-6c9e-4112-80d0-0a6c76df1643.png Ig woh 4 mentors wala column rh gya h

This pr is not completed yet, I am working on home page.
I will create that column.

**Comment by anmode** [comment]
ohh ok but for the service section can you give the card to the same property as we have on the card before.. it was looking awesome.

**Comment by Isha988** [comment]
> ohh ok but for the service section can you give the card to the same property as we have on the card before.. it was looking awesome. Screenshot 2023-07-13 at 11 51 01 AM

okay, I will

**Comment by Isha988** [comment]
@anmode have a look at internship section

**Comment by Isha988** [comment]
@amode only banner and service section are remaining, rest of the sections are completed.
Hackathon section is not looking that good, suggest me some idea for that.
If anything else is not looking up to the mark, so please let me know.

**Comment by anmode** [comment]
> @amode only banner and service section are remaining, rest of the sections are completed. Hackathon section is not looking that good, suggest me some idea for that. If anything else is not looking up to the mark, so please let me know.

wow, isha nice changes so far,
for the hackathon section ( home page) - No worries for mobile mode just reduce card size, two cards at one level. would be fine. In desktop mode it is ok

View more hackathon section distorted somehow.... No worries I'll give that section to somebody to fix it. You can focus on the rest part of the homepage.

Did you receive the email?? I sent you yesterday

**Review by anmode — COMMENTED** [review]
Do one thing.. MentorSection , testimonial and other related to homepage... It is better to keep them in components folder inside Homepage folder inside the component. ( `components/hompage/components`

**Comment by Isha988** [comment]
Banner is almost ready, just new image have to be added.

**Comment by Isha988** [comment]
> ohh ok but for the service section can you give the card to the same property as we have on the card before.. it was looking awesome. Screenshot 2023-07-13 at 11 51 01 AM

Is service looking good now, or I change it to the previous version.

**Comment by anmode** [comment]
> @anmode Is this image looking okay? Should I commit these changes? Screenshot 2023-07-14 at 13-59-29 GrabTern

yess it is great

**Comment by anmode** [comment]
> > ohh ok but for the service section can you give the card to the same property as we have on the card before.. it was looking awesome. Screenshot 2023-07-13 at 11 51 01 AM
> 
> Is service section looking good now as it is matching theme, or I change it to the previous version.

.

**Comment by anmode** [comment]
Screenshot 2023-07-14 at 4 43 44 PM
Isha I noticed a black border in your image look bulb vector... not looking nice. Please on those vector revert back to original

**Comment by Isha988** [comment]
> Screenshot 2023-07-14 at 4 43 44 PM Isha I noticed a black border in your image look bulb vector... not looking nice. Please on those vector revert back to original

If you see the very first deployment of this pr the border exist, even I had made any changes in banner at that time.

**Comment by anmode** [comment]
> > Screenshot 2023-07-14 at 4 43 44 PM Isha I noticed a black border in your image look bulb vector... not looking nice. Please on those vector revert back to original
> 
> If you see the very first deployment of this pr the border exist, even I had made any changes in banner at that time. This may be because we are using emoji instead of image/svg and maybe it is appearing different in different systems.

oh ok. I'll check it... please commit your changes. PR is good to complete.

**Comment by Isha988** [comment]
> > > Screenshot 2023-07-14 at 4 43 44 PM Isha I noticed a black border in your image look bulb vector... not looking nice. Please on those vector revert back to original
> > 
> > 
> > If you see the very first deployment of this pr the border exist, even I had made any changes in banner at that time. This may be because we are using emoji instead of image/svg and maybe it is appearing different in different systems.
> 
> oh ok. I think your statement is correct may this is because of emojies use... please commit your changes. PR is good to complete.

Just wait, working on service section.

File: components/homePage/Banner.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+function Banner({ isMentorLoggedIn }) {
+  return (
+    <div className="tw-min-h-screen tw-w-full tw-overflow-hidden">
+      <div className="tw-min-h-screen tw-w-full tw-max-w-7xl tw-mx-auto tw-flex tw-items-center tw-px-4">
+        <div className="tw-w-full tw-grid tw-grid-cols-12">
+          <div className="tw-col-span-12 lg:tw-col-span-5">
+            <div className="tw-h-full tw-flex tw-items-center tw-mb-8 lg:tw-mb-0">
+              {/* text box */}
+              <div className="tw-mt-12 tw-text-center lg:tw-text-left tw-font-sans tw-space-y-4">
+                <h1 className="tw-font-medium tw-text-4xl lg:tw-text-5xl !tw-leading-[60px] tw-font-[Poppins]">
```
Comment (anmode) [inline components/homePage/Banner.js:13]: I just checked it needs margin in mobile mode `mt-4 lg:mt-0`. Just see mobile mode.

File: components/layout/Header.js [hunk 2]
```diff
@@ -58,27 +59,29 @@ function Header() {
...
+      <nav className="tw-w-full tw-max-w-7xl tw-mx-auto tw-flex tw-justify-between tw-items-center ">
+        <Link
+          href="/"
+          className="tw-flex tw-gap-2 tw-justify-center tw-items-center"
+        >
+          <Image
+            src={logo}
+            alt="grabtern_logo"
+            width={85}
+            height={85}
```
Comment (anmode) [inline components/layout/Header.js:79]: Could you try to increase the size of logo?  chota sa lg rha h logo boht :(

File: components/layout/Header.js [hunk 3]
```diff
@@ -89,96 +92,97 @@ function Header() {
...
-                className=" tw-w-full tw-rounded-lg tw-border btn:hover btn:active tw-bg-[#845ec2] hover:tw-scale-[0.99] active:tw-scale-[1.01] active:tw-shadow-none tw-py-3 tw-text-white tw-transition-all tw-text-center tw-text-md tw-font-inter tw-flex tw-items-center tw-justify-center"
-              >
-                Signup
-              </button>
-
-              {/* <DropdownCard
+        </div>
+
+        {/* For Mobile Navigation */}
+        <div className="sm:tw-hidden tw-flex tw-relative" ref={dropdownRef}>
```
Comment (anmode) [inline components/layout/Header.js:130]: need to be changed here, If you clearly see when the dropdown opens it does not converge the full width of the device.

File: components/layout/Header.js [hunk 4]
```diff
@@ -89,96 +92,97 @@ function Header() {
...
+            <div
+              className="tw-text-3xl tw-font-light tw-mx-2 cursor-pointer tw-text-[#845ec2]"
+              onClick={() => {
+                setToggleDropdown((prev) => !prev);
+              }}
+            >
+              <ion-icon name={toggleDropdown ? "close" : "menu"}></ion-icon>
+            </div>
+            {toggleDropdown && (
+              <div className="tw-absolute tw-right-0 tw-top-full tw-mt-0 tw-w-full tw-p-6 tw-rounded-lg tw-bg-white tw-min-w-[100vw] tw-flex tw-flex-col tw-gap-5 tw-justify-center tw-items-center tw-text-sm tw-font-semibold tw-pb-5">
```
Comment (anmode) [inline components/layout/Header.js:141]: may be here `tw-w-screen` replace with `tw-min-w-[100vw]`

**Review by anmode — CHANGES_REQUESTED** [review]
Some comments all good. Really good work I must say. I need to learn more now :)

**Comment by anmode** [comment]
Screenshot 2023-07-15 at 1 36 36 PM
width ko 100% se jyda krne pr theek behaviour tha.. this mean koi base property theek ni h

**Comment by Isha988** [comment]
> Just wait, working on service section.

I have reverted the service section changes, but It is not producing same result.
Is any changes are made in css files?

File: components/layout/Header.js (hunk 4 above)
Comment (Isha988) [reply components/layout/Header.js]: done 👍

**Comment by Isha988** [comment]
it is fixed navbar width.

File: components/homePage/Banner.js (hunk 1 above)
Comment (Isha988) [reply components/homePage/Banner.js:13]: I already added tw-mt-12 for that.

**Comment by anmode** [comment]
> > Just wait, working on service section.
> 
> I have reverted the service section changes, but It is not producing same result. Is any changes are made in css files?

No, I hope.

**Comment by anmode** [comment]
I see,
1. Vector used by you before were good. Could you replace vector now with your previous vector in service section?
2. For nav, Actually scrolling behaviour of nav bar was looking good. and please as now all things complete could you remove the shadow effect in nav bar?
Rest all is awesome

**Comment by Isha988** [comment]
@amode please give a final review with all points in your haed. I am getting confused with so many comments.

**Comment by anmode** [comment]
> @amode please give a final review with all points in your head. I am getting confused with so many comments.

oh okay sure, Vercel did a great thing. we can now talk in preview and I will comment about UI there and in github code improvement is that ok?

**Comment by Isha988** [comment]
> > @amode please give a final review with all points in your head. I am getting confused with so many comments.
> 
> oh okay sure, Vercel did a great thing. we can now talk in preview mode and I will comment about UI there and in github, code improvement is that ok?

ya sure

**Comment by Isha988** [comment]
> 1. Vector used by you before were good. Could you replace vector now with your previous vector in service section?

Should I change the complete design or just Icons
Screenshot 2023-07-15 at 15-32-46 GrabTern

**Comment by anmode** [comment]
> > 1. Vector used by you before were good. Could you replace vector now with your previous vector in service section?
> 
> Should I change the complete design or just Icons
> Screenshot 2023-07-15 at 15-32-46 GrabTern
> 
Just iconss

**Comment by Isha988** [comment]
> Just iconss
Screenshot 2023-07-15 at 15-41-27 GrabTern

now?

**Comment by anmode** [comment]
yess all cool now!

**Comment by Isha988** [comment]
> yess all cool now!

Anything else?

**Comment by anmode** [comment]
> > yess all cool now!
> 
> Anything else?

Nope,

**Review by anmode — APPROVED** [review]
Cool!

**Comment by Isha988** [comment]
> Cool!

Finally 🚀. Really a big one

**Comment by anmode** [comment]
One thing your Pic is missing in the grabtern community :)! If you agree. you can add.
Merging this PR as of now!
Thanks for making refactoring of the Homepage!!

**Comment by Isha988** [comment]
> One thing your Pic is missing in the grabtern community :)! If you agree. you can add. Merging this PR as of now! Thanks for making refactoring of the Homepage!!

I profile is in top contributors of Grabtern, that's is enough.✌️

**Comment by anmode** [comment]
> > One thing your Pic is missing in the grabtern community :)! If you agree. you can add. Merging this PR as of now! Thanks for making refactoring of the Homepage!!
> 
> I profile is in top contributors of Grabtern, that's is enough.✌️

yeah, that's too! :)

**Comment by anmode** [comment]
Reminding again: I sent you the mail! updated agreement. 
If you find it suitable you can reply. 
thanks

**Pull request merged**